pkg/prm/exec_test.go

```diff
@@ -0,0 +1,144 @@
+package prm_test
+
+import (
+	"testing"
+
+	"github.com/Masterminds/semver"
+	"github.com/mitchellh/mapstructure"
+	"github.com/puppetlabs/prm/internal/pkg/mock"
+	"github.com/puppetlabs/prm/pkg/prm"
+)
+
+func TestPrm_Exec(t *testing.T) {
+	tests := []struct {
+		name        string
+		expectError bool
```

Comment:
I think the main thing we want to check here is that the error message is correct, so maybe `expectedErrorMessage` and then if it returns an error and `expectedErrorMessage` is not nil then we want to assert that the error message matches.

As identified during pairing today, the messages may well be _wrong_ but we can ticket that out for `0.2.0`.

This is a strong suggestion but should not block the release path for 0.1.0.

Reply:
Updated to use `expectedErrMsg` instead of `expectError`.
For the errors that aren't returning as they should have left the error we would want returned in a comment to the side of each variable.